concrete/src/Http/Middleware/ThumbnailMiddleware.php

```diff
@@ -86,32 +92,77 @@ public function process(Request $request, DelegateInterface $frame)
     }
 
     /**
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    private function buildRedirectToThumbnailResponse(Request $request)
+    {
+        $redirectTo = (string) $request->getUri();
+        $redirectTo .= ((strpos($redirectTo, '?') === false) ? '?' : '&') . mt_rand();
+        $responseFactory = $this->app->make(ResponseFactoryInterface::class);
+        $response = $responseFactory->redirect($redirectTo);
+
+        return $response;
+    }
+
+    /**
+     * @param Connection $database
+     * @param string|null $searchThumbnailPath
+     *
+     * @return ConcretePDOStatement|null
```

Comment:
This should just return an `\Iterable` (so `string[]|\Iterator`), we should avoid returning nullable types, especially when dealing with lists.